core/__tests__/models/mapping.ts

```diff
@@ -66,7 +72,7 @@ describe("models/mapping", () => {
           propertyId: userId.id,
           remoteKey: "bar",
         })
-      ).rejects.toThrow(/There is already a Mapping for/);
+      ).rejects.toThrow(/Validation error/);
```

Comment:
DB catches this now. I added a test below specifically for the new afterSave logic.